javascriptv3/example_code/cross-services/transcribe-streaming-app/src/index.js

```diff
@@ -13,189 +13,91 @@ index.js is part of a tutorial demonstrating how to:
-    client.destroy();
-  } catch (err) {
-    console.log("Error. ", err);
+    await TranscribeClient.startRecording(selectedLanguage, onTranscriptionDataReceived);
+  } catch(error) {
+    alert("An error occurred while recording: " + error.message);
+    stopRecording();
   }
 };
 
-window.stopRecord = function () {
-  window.location.reload();
+const onTranscriptionDataReceived = (data) => {
+  const outputDiv = document.getElementById("output");
+  outputDiv.insertAdjacentHTML("beforeend", data);
+}
+
+const stopRecording = function () {
+  let recordingButton = document.getElementById("record");
+  let languageList = document.getElementById("inputLanguageList");
+  languageList.disabled = false;
+  recordingButton.setAttribute("class", "recordInactive");
+  TranscribeClient.stopRecording();
 };
 
 window.translateText = async () => {
+  const sourceText = document.getElementById("output").innerHTML;
+  if (sourceText.length === 0) {
+    alert("No text to translate!");
+    return;
+  }
+  const targetLanguage = document.getElementById("translationLanguageList").value;
+  if (targetLanguage === "nan") {
+    alert("Please select a language to translate to!");
+    return;
+  }
   try {
-    const outPut = document.getElementById("output").innerHTML;
-    const data = await comprehendClient.send(
-      new DetectDominantLanguageCommand({ Text: outPut })
-    );
-    // return data; // Uncomment this for unit tests.
-    const langCode = data.Languages[0].LanguageCode;
-    try {
-      const selectedValue = document.getElementById("list").value;
-      const translateParams = {
-        Text: outPut,
-        SourceLanguageCode: langCode /* required */,
-        TargetLanguageCode: selectedValue /* required */,
-      };
-      const data = await translateClient.send(
-        new TranslateTextCommand(translateParams)
-      );
-      document.getElementById("translated").innerHTML = data.TranslatedText;
-    } catch (err) {
-      console.log("Error translating language. ", err);
+    let translation = await TranslateClient.translateTextToLanguage(sourceText, targetLanguage);
+    if (translation) {
+      document.getElementById("translated").innerHTML = translation;
     }
-  } catch (err) {
-    console.log("Error detecting language of text. ", err);
+  } catch (error) {
+    alert("There was an error translating the text: " + error.message);
   }
 };
 
-window.clearTranscription = async () => {
+window.clearTranscription = () => {
   document.getElementById("output").innerHTML = "";
+  document.getElementById("translated").innerHTML = "";
 };
 
-// Helper function to send an email to user.
 window.sendEmail = async () => {
-  const toEmail = document.getElementById("email").value;
-  const outputDiv = document.getElementById("output").innerHTML;
-  const translatedDiv = document.getElementById("translated").innerHTML;
-  const fromEmail = "SENDER_ADDRESS";
+  const receiver = document.getElementById("email").value;
```

Comment:
This could be renamed `email` and used below instead of setting `const sender = receiver`

Reply:
The way it was originally written, I found it a bit hard to understand what was going on. The sender email was just a plain string, and needed to be set in code. Which I thought was not a good thing. So I changed the logic so that an email was sent from the account in the form field to that same account. One less thing to be set by the end user, and the code makes it obvious what's going on.